apps/omg_watcher/test/db/txoutput_db_test.exs

```diff
@@ -0,0 +1,241 @@
+          ],
+          number: 1000
+        })
+
+      assert %TxOutputDB{
+               owner: ^alice_addr,
+               currency: @eth,
+               amount: 2,
+               creating_deposit: nil,
+               creating_txhash: ^txhash2,
+               creating_tx_oindex: 1
+             } = TxOutputDB.get_by_position(Utxo.position(1000, 1, 1))
+
+      [{:ok, _tx0}, {:ok, _tx1}, {:ok, %TransactionDB{txhash: txhash3}}] =
+        TransactionDB.update_with(%Block{
+          transactions: [
+            API.TestHelper.create_recovered([], @eth, [{alice, 20_131}]),
+            API.TestHelper.create_recovered([], @eth, [{alice, 20_132}]),
+            API.TestHelper.create_recovered([], @eth, [{alice, 3}, {alice, 4}])
+          ],
+          number: 2013
+        })
+
+      assert %TxOutputDB{
+               owner: ^alice_addr,
+               currency: @eth,
+               amount: 3,
+               creating_deposit: nil,
+               creating_txhash: ^txhash3,
+               creating_tx_oindex: 0
+             } = TxOutputDB.get_by_position(Utxo.position(2013, 2, 0))
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox]
+    test "create outputs: creates proper transaction's outputs" do
+      newowner1 = <<1::160>>
+      amount1 = 1
+      newowner2 = <<2::160>>
+      amount2 = 2
+
+      tx = %Transaction{cur12: @eth, newowner1: newowner1, amount1: amount1, newowner2: newowner2, amount2: amount2}
+
+      [utxo1, utxo2] = TxOutputDB.create_outputs(tx)
+
+      assert %TxOutputDB{owner: newowner1, amount: amount1, currency: @eth, creating_tx_oindex: 0} == utxo1
+
+      assert %TxOutputDB{owner: newowner2, amount: amount2, currency: @eth, creating_tx_oindex: 1} == utxo2
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox]
+    test "create outputs: output with zero amount is skipped" do
+      newowner1 = <<1::160>>
+      amount1 = 1
+      newowner2 = <<0::160>>
+      amount2 = 0
+
+      tx = %Transaction{cur12: @eth, newowner1: newowner1, amount1: amount1, newowner2: newowner2, amount2: amount2}
+
+      [utxo1] = TxOutputDB.create_outputs(tx)
+
+      assert %TxOutputDB{owner: newowner1, amount: amount1, currency: @eth, creating_tx_oindex: 0} == utxo1
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox, :alice]
+    test "get inputs: prepares existing utxo for spend", %{alice: alice} do
+      [{:ok, _evnt}] =
+        EthEventDB.insert_deposits([%{blknum: 1, owner: alice.addr, currency: @eth, amount: 1, hash: "hash1"}])
+
+      tx = %Transaction{blknum1: 1, txindex1: 0, oindex1: 0, blknum2: 0, txindex2: 0, oindex2: 0}
+      [changeset] = TxOutputDB.get_inputs(tx)
+
+      assert %Ecto.Changeset{
+               data: %TxOutputDB{creating_deposit: "hash1", spending_tx_oindex: nil},
+               changes: %{spending_tx_oindex: 0}
+             } = changeset
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox]
+    test "get inputs: providing non-existing utxo position results in empty list" do
```

Comment:
Probably we should be more defensive and return error when inputs to tx can't be fond... For the other hand it will complicates tests (a little) and I'm not sure how well it'll match with checkpoints feature